### PR TITLE
Avoid call strlen repeatedly in loop.

### PR DESCRIPTION
**Pull request by raniervf**
##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: ssl/ssl_rsa.c [hunk 1]
```diff
@@ -916,6 +916,9 @@ int SSL_CTX_use_serverinfo_file(SSL_CTX *ctx, const char *file)
     char *header = NULL;
     char namePrefix1[] = "SERVERINFO FOR ";
     char namePrefix2[] = "SERVERINFOV2 FOR ";
+    unsigned int namePrefix1_len;
```
Comment (InfoHunter) [inline ssl/ssl_rsa.c]: you can just use `sizeof(namePrefix1) - 1` here and avoid more `strlen`s

File: ssl/ssl_rsa.c [hunk 2]
```diff
@@ -951,19 +956,20 @@ int SSL_CTX_use_serverinfo_file(SSL_CTX *ctx, const char *file)
                 break;
         }
         /* Check that PEM name starts with "BEGIN SERVERINFO FOR " */
-        if (strlen(name) < strlen(namePrefix1)) {
+        name_len = strlen(name);
+        if (name_len < namePrefix1_len) {
             SSLerr(SSL_F_SSL_CTX_USE_SERVERINFO_FILE, SSL_R_PEM_NAME_TOO_SHORT);
             goto end;
         }
-        if (strncmp(name, namePrefix1, strlen(namePrefix1)) == 0) {
```
Comment (InfoHunter) [inline ssl/ssl_rsa.c]: Or you can replace `strlen()` into `sizeof` here to avoid introducing more local variables.

**Comment by raniervf** [comment]
The declaration char namePrefix1[] = "SERVERINFO FOR ";, , leave doubts, if original author,
will introduce namePrefix mutable in future.

**Comment by mspncp** [comment]
Joined with #10380.

**Comment by mspncp** [comment]
> The declaration char namePrefix1[] = "SERVERINFO FOR ";, , leave doubts, if original author,
> will introduce namePrefix mutable in future.

That's why I would make it a `static const char namePrefix1[]`

File: ssl/ssl_rsa.c (hunk 1 above)
Comment (richsalz) [reply ssl/ssl_rsa.c]: the variables should be "static"

**Pull request closed without merging**